DuckDuckGo/NavigationBar/View/ShowToolbarsOnFullScreenMenuFactory.swift

```diff
@@ -0,0 +1,45 @@
+//
+//  ShowToolbarsOnFullScreenMenuFactory.swift
+//
+//  Copyright © 2024 DuckDuckGo. All rights reserved.
+//
+//  Licensed under the Apache License, Version 2.0 (the "License");
+//  you may not use this file except in compliance with the License.
+//  You may obtain a copy of the License at
+//
+//  http://www.apache.org/licenses/LICENSE-2.0
+//
+//  Unless required by applicable law or agreed to in writing, software
+//  distributed under the License is distributed on an "AS IS" BASIS,
+//  WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+//  See the License for the specific language governing permissions and
+//  limitations under the License.
+//
+
+import Foundation
+import AppKit
+
+struct ShowToolbarsOnFullScreenMenuFactory {
```

Comment:
Maybe `ShowToolbarsOnFullScreenMenuCoordinator` would be a better name? This essentially manages a single menu item, rather than producing them.